frontend/src/store/objectStore.ts

```diff
@@ -82,8 +86,12 @@ export const useObjectStore = defineStore('objectStore', () => {
             obj.metadata = metadata;
             obj.name = metadata.metadata.find((x: { key: string }) => x.key === 'name')?.value;
           }
-        });
 
+          // Add the permissions to each object list item
```

Comment:
i think in line 77 we are removing the perms and then adding them again here in line 91. not sure if that could be optimized.

Reply:
I think the line 77 being referred to is just stashing the specific IDs from the permissions to fetch the objects? So not removing them?
(file has changed since so line 77 might be something else from original comment)

Reply:
my point was: we are looping through permResponse in line 77, the perms can be added there instead of doing another loop. only mention this because for a bucket full of objects this could impact the speed